aws/resource_aws_acm_certificate.go

```diff
@@ -275,16 +277,43 @@ func resourceAwsAcmCertificateUpdate(d *schema.ResourceData, meta interface{}) e
 	return resourceAwsAcmCertificateRead(d, meta)
 }
 
-func cleanUpSubjectAlternativeNames(cert *acm.CertificateDetail) []string {
-	sans := cert.SubjectAlternativeNames
-	vs := make([]string, 0)
-	for _, v := range sans {
-		if aws.StringValue(v) != aws.StringValue(cert.DomainName) {
-			vs = append(vs, aws.StringValue(v))
+// readSubjectAlternativeNames returns all subject alternative names (SANs)
+// fetched from remote in cert but ordered according to user-defined state.
+// Also, the domain_name is excluded from the result.
+//
+// Sorting keeps the state stable and is necessary because the API may return
+// SANS ordred differetnly from request to request.
+func readSubjectAlternativeNames(cert *acm.CertificateDetail, d *schema.ResourceData) []string {
+	remotes := cert.SubjectAlternativeNames
+	sans := make([]string, 0)
+
+	// intersection of all SANs, user defined (ud) and remote. ordered as user defined (ud) in state.
+	for _, ud := range d.Get("subject_alternative_names").([]interface{}) {
```

Comment:
I like this idea to have the certificates ordered as defined by user.
Just want to highlight that when I tried to use this solution (with terraform 0.11.4 and based on terraform aws provider 2.23) SANs were not ordered as defined by user because `d.Get("subject_alternative_names")` returned an empty list during the refresh.

Reply:
I tried to use this PR as well, and had issues with it.  I have created a new pull request that should address the problems: #10791